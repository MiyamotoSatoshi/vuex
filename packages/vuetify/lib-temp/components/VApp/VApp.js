import '../../stylus/components/_app.styl';
// Component level mixins
import AppTheme from './mixins/app-theme';
import Themeable from '../../mixins/themeable';
// Directives
import Resize from '../../directives/resize';
/* @vue/component */
export default {
    name: 'v-app',
    directives: {
        Resize
    },
    mixins: [
        AppTheme,
        Themeable
    ],
    props: {
        id: {
            type: String,
            default: 'app'
        },
        dark: Boolean
    },
    computed: {
        classes() {
            return {
                'application--is-rtl': this.$vuetify.rtl,
                ...this.themeClasses
            };
        }
    },
    watch: {
        dark() {
            this.$vuetify.dark = this.dark;
        }
    },
    mounted() {
        this.$vuetify.dark = this.dark;
    },
    render(h) {
        const data = {
            staticClass: 'application',
            'class': this.classes,
            attrs: { 'data-app': true },
            domProps: { id: this.id }
        };
        const wrapper = h('div', { staticClass: 'application--wrap' }, this.$slots.default);
        return h('div', data, [wrapper]);
    }
};
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiVkFwcC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIi4uLy4uLy4uL3NyYy9jb21wb25lbnRzL1ZBcHAvVkFwcC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLG1DQUFtQyxDQUFBO0FBRTFDLHlCQUF5QjtBQUN6QixPQUFPLFFBQVEsTUFBTSxvQkFBb0IsQ0FBQTtBQUV6QyxPQUFPLFNBQVMsTUFBTSx3QkFBd0IsQ0FBQTtBQUU5QyxhQUFhO0FBQ2IsT0FBTyxNQUFNLE1BQU0seUJBQXlCLENBQUE7QUFFNUMsb0JBQW9CO0FBQ3BCLGVBQWU7SUFDYixJQUFJLEVBQUUsT0FBTztJQUViLFVBQVUsRUFBRTtRQUNWLE1BQU07S0FDUDtJQUVELE1BQU0sRUFBRTtRQUNOLFFBQVE7UUFDUixTQUFTO0tBQ1Y7SUFFRCxLQUFLLEVBQUU7UUFDTCxFQUFFLEVBQUU7WUFDRixJQUFJLEVBQUUsTUFBTTtZQUNaLE9BQU8sRUFBRSxLQUFLO1NBQ2Y7UUFDRCxJQUFJLEVBQUUsT0FBTztLQUNkO0lBRUQsUUFBUSxFQUFFO1FBQ1IsT0FBTztZQUNMLE9BQU87Z0JBQ0wscUJBQXFCLEVBQUUsSUFBSSxDQUFDLFFBQVEsQ0FBQyxHQUFHO2dCQUN4QyxHQUFHLElBQUksQ0FBQyxZQUFZO2FBQ3JCLENBQUE7UUFDSCxDQUFDO0tBQ0Y7SUFFRCxLQUFLLEVBQUU7UUFDTCxJQUFJO1lBQ0YsSUFBSSxDQUFDLFFBQVEsQ0FBQyxJQUFJLEdBQUcsSUFBSSxDQUFDLElBQUksQ0FBQTtRQUNoQyxDQUFDO0tBQ0Y7SUFFRCxPQUFPO1FBQ0wsSUFBSSxDQUFDLFFBQVEsQ0FBQyxJQUFJLEdBQUcsSUFBSSxDQUFDLElBQUksQ0FBQTtJQUNoQyxDQUFDO0lBRUQsTUFBTSxDQUFFLENBQUM7UUFDUCxNQUFNLElBQUksR0FBRztZQUNYLFdBQVcsRUFBRSxhQUFhO1lBQzFCLE9BQU8sRUFBRSxJQUFJLENBQUMsT0FBTztZQUNyQixLQUFLLEVBQUUsRUFBRSxVQUFVLEVBQUUsSUFBSSxFQUFFO1lBQzNCLFFBQVEsRUFBRSxFQUFFLEVBQUUsRUFBRSxJQUFJLENBQUMsRUFBRSxFQUFFO1NBQzFCLENBQUE7UUFFRCxNQUFNLE9BQU8sR0FBRyxDQUFDLENBQUMsS0FBSyxFQUFFLEVBQUUsV0FBVyxFQUFFLG1CQUFtQixFQUFFLEVBQUUsSUFBSSxDQUFDLE1BQU0sQ0FBQyxPQUFPLENBQUMsQ0FBQTtRQUVuRixPQUFPLENBQUMsQ0FBQyxLQUFLLEVBQUUsSUFBSSxFQUFFLENBQUMsT0FBTyxDQUFDLENBQUMsQ0FBQTtJQUNsQyxDQUFDO0NBQ0YsQ0FBQSIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCAnLi4vLi4vc3R5bHVzL2NvbXBvbmVudHMvX2FwcC5zdHlsJ1xuXG4vLyBDb21wb25lbnQgbGV2ZWwgbWl4aW5zXG5pbXBvcnQgQXBwVGhlbWUgZnJvbSAnLi9taXhpbnMvYXBwLXRoZW1lJ1xuXG5pbXBvcnQgVGhlbWVhYmxlIGZyb20gJy4uLy4uL21peGlucy90aGVtZWFibGUnXG5cbi8vIERpcmVjdGl2ZXNcbmltcG9ydCBSZXNpemUgZnJvbSAnLi4vLi4vZGlyZWN0aXZlcy9yZXNpemUnXG5cbi8qIEB2dWUvY29tcG9uZW50ICovXG5leHBvcnQgZGVmYXVsdCB7XG4gIG5hbWU6ICd2LWFwcCcsXG5cbiAgZGlyZWN0aXZlczoge1xuICAgIFJlc2l6ZVxuICB9LFxuXG4gIG1peGluczogW1xuICAgIEFwcFRoZW1lLFxuICAgIFRoZW1lYWJsZVxuICBdLFxuXG4gIHByb3BzOiB7XG4gICAgaWQ6IHtcbiAgICAgIHR5cGU6IFN0cmluZyxcbiAgICAgIGRlZmF1bHQ6ICdhcHAnXG4gICAgfSxcbiAgICBkYXJrOiBCb29sZWFuXG4gIH0sXG5cbiAgY29tcHV0ZWQ6IHtcbiAgICBjbGFzc2VzICgpIHtcbiAgICAgIHJldHVybiB7XG4gICAgICAgICdhcHBsaWNhdGlvbi0taXMtcnRsJzogdGhpcy4kdnVldGlmeS5ydGwsXG4gICAgICAgIC4uLnRoaXMudGhlbWVDbGFzc2VzXG4gICAgICB9XG4gICAgfVxuICB9LFxuXG4gIHdhdGNoOiB7XG4gICAgZGFyayAoKSB7XG4gICAgICB0aGlzLiR2dWV0aWZ5LmRhcmsgPSB0aGlzLmRhcmtcbiAgICB9XG4gIH0sXG5cbiAgbW91bnRlZCAoKSB7XG4gICAgdGhpcy4kdnVldGlmeS5kYXJrID0gdGhpcy5kYXJrXG4gIH0sXG5cbiAgcmVuZGVyIChoKSB7XG4gICAgY29uc3QgZGF0YSA9IHtcbiAgICAgIHN0YXRpY0NsYXNzOiAnYXBwbGljYXRpb24nLFxuICAgICAgJ2NsYXNzJzogdGhpcy5jbGFzc2VzLFxuICAgICAgYXR0cnM6IHsgJ2RhdGEtYXBwJzogdHJ1ZSB9LFxuICAgICAgZG9tUHJvcHM6IHsgaWQ6IHRoaXMuaWQgfVxuICAgIH1cblxuICAgIGNvbnN0IHdyYXBwZXIgPSBoKCdkaXYnLCB7IHN0YXRpY0NsYXNzOiAnYXBwbGljYXRpb24tLXdyYXAnIH0sIHRoaXMuJHNsb3RzLmRlZmF1bHQpXG5cbiAgICByZXR1cm4gaCgnZGl2JywgZGF0YSwgW3dyYXBwZXJdKVxuICB9XG59XG4iXX0=