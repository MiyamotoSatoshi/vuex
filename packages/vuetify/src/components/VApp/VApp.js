// Styles
import '../../stylus/components/_app.styl'

// Mixins
import Themeable from '../../mixins/themeable'

/* @vue/component */
export default {
  name: 'v-app',

  mixins: [Themeable],

  props: {
    dark: {
      type: Boolean,
      default: undefined
    },
    id: {
      type: String,
      default: 'app'
    },
    light: {
      type: Boolean,
      default: undefined
    }
  },

  computed: {
    classes () {
      return {
        'application--is-rtl': this.$vuetify.rtl,
        ...this.themeClasses
      }
    },
    isDark () {
      if (this.dark != null || this.light != null) {
        return Themeable.options.computed.isDark.call(this)
      }

      return this.$vuetify.theme.dark
    }
  },

  render (h) {
    const data = {
      staticClass: 'application',
      'class': this.classes,
      attrs: { 'data-app': true },
      domProps: { id: this.id }
    }

    const wrapper = h('div', { staticClass: 'application--wrap' }, this.$slots.default)

    return h('div', data, [wrapper])
  }
}
