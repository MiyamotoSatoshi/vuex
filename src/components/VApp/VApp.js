require('../../stylus/components/_app.styl')

import Breakpoint from '../../util/breakpoint'
import Themeable from '../../mixins/themeable'

import Resize from '../../directives/resize'

export default {
  name: 'v-app',

  mixins: [Breakpoint, Themeable],

  directives: {
    Resize
  },

  data: () => ({
    resizeTimeout: {}
  }),

  props: {
    id: {
      type: String,
      default: 'app'
    }
  },

  mounted () {
    this.onResize()
  },

  render (h) {
    const data = {
      staticClass: 'application',
      'class': {
        'application--dark': this.dark,
        'application--light': !this.dark
      },
      attrs: { 'data-app': true },
      domProps: { id: this.id },
      directives: [{
        name: 'resize',
        value: this.onResize
      }]
    }

    return h('div', data, this.$slots.default)
  }
}
