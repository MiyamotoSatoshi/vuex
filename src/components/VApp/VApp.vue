<script>
  import Breakpoint from '../../util/breakpoint'
  import Themeable from '../../mixins/themeable'

  import Resize from '../../directives/resize'

  export default {
    name: 'v-app',

    mixins: [Breakpoint, Themeable],

    directives: {
      Resize
    },

    data: () => ({
      resizeTimeout: {}
    }),

    props: {
      fixedFooter: Boolean,
      footer: Boolean,
      id: {
        type: String,
        default: 'app'
      },
      toolbar: Boolean
    },

    methods: {
      resizeDebounce () {
        clearTimeout(this.resizeTimeout)

        this.resizeTimeout = setTimeout(this.onResize, 200)
      }
    },

    mounted () {
      this.onResize()
    },

    render (h) {
      const data = {
        staticClass: 'application',
        'class': {
          'application--dark': this.dark,
          'application--light': !this.dark,
          'application--footer': this.footer || this.fixedFooter,
          'application--footer-fixed': this.fixedFooter,
          'application--toolbar': this.toolbar
        },
        attrs: { 'data-app': true },
        domProps: { id: this.id },
        directives: [{
          name: 'resize',
          value: this.resizeDebounce
        }]
      }

      return h('div', data, this.$slots.default)
    }
  }
</script>

<style lang="stylus" src="../../stylus/components/_app.styl"></style>
