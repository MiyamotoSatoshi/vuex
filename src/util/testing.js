import Vue from 'vue'
import load from '~util/load'
import { mount, shallow } from 'avoriaz'

export function test(name, cb) {
  Vue.prototype.$vuetify = { load }

  beforeEach(() => {
    rafPolyfill(window)
  })

  describe(name, () => cb({
    functionalContext,
    mount,
    shallow
  }))
}

export function functionalContext(context = {}) {
  return {
    context: Object.assign({
      data: {},
      props: {}
    }, context)
  }
}


//requestAnimationFrame polyfill | Milos Djakonovic ( @Miloshio ) | MIT | https://github.com/milosdjakonovic/requestAnimationFrame-polyfill
export function rafPolyfill(w) {
  /**
   *
   * How many times should polyfill call
   * update callback? By canon, it should
   * be 60 times per second, so that ideal
   * framerate 60fps could be reached.
   *
   * However, even native implementations
   * of requestAnimationFrame often cannot
   * do 60fps, but, unlike any polyfill,
   * they can synchronise achievable fps
   * rate with screen refresh rate.
   *
   * So, leave this value 1000/60 unless
   * you target specific browser on spec
   * ific device that is going to work
   * better with custom value. I think
   * that this is the longest comment I've
   * written on single variable so far.
  **/
  var FRAME_RATE_INTERVAL = 1000/60,

  /**
   * All queued callbacks in given cycle.
  **/
  allCallbacks = [],

  executeAllScheduled = false,

  shouldCheckCancelRaf = false,

  /**
   * Callbacks queued for cancellation.
  **/
  callbacksForCancellation = [],

  /**
   * Should callback be cancelled?
   * @param cb - callback
  **/
  isToBeCancelled = function(cb){
    for(var i=0;i<callbacksForCancellation.length;i++){
      if(callbacksForCancellation[i] === cb ){
        callbacksForCancellation.splice(i,1);
        return true;
      }
    }
  },



  /**
   *
   * Executes all (surprise) callbacks in
   * and removes them from callback queue.
   *
  **/
  executeAll = function(){
    executeAllScheduled = false;
    var _allCallbacks = allCallbacks;
    allCallbacks = [];
    for(var i=0;i<_allCallbacks.length;i++){
      if(shouldCheckCancelRaf===true){
        if (isToBeCancelled(_allCallbacks[i])){
          shouldCheckCancelRaf = false;
          return;
        }
      }
      _allCallbacks[i].apply(w, [ new Date().getTime() ] );
    }
  },

  /**
   *
   * requestAnimationFrame polyfill
   * @param callback - callback to be queued & executed | executed
   * @return callback
   *
  **/
  raf = function(callback){
    allCallbacks.push(callback);
    if(executeAllScheduled===false){
      w.setTimeout(executeAll, FRAME_RATE_INTERVAL);
      executeAllScheduled = true;
    }
    return callback;
  },

  /**
   *
   * Cancels raf.
  **/
  cancelRaf = function(callback){
    callbacksForCancellation.push(callback);
    shouldCheckCancelRaf = true;
  },


  //https://gist.github.com/paulirish/1579671
  vendors = ['ms', 'moz', 'webkit', 'o'];

    for(var x = 0; x < vendors.length && !w.requestAnimationFrame; ++x) {
        w.requestAnimationFrame = w[vendors[x]+'RequestAnimationFrame'];
        w.cancelAnimationFrame = w[vendors[x]+'CancelAnimationFrame']
        || w[vendors[x]+'CancelRequestAnimationFrame'];
    }

  if (!w.requestAnimationFrame) w.requestAnimationFrame = raf;
  if (!w.cancelAnimationFrame)  w.cancelAnimationFrame  = cancelRaf;

}
