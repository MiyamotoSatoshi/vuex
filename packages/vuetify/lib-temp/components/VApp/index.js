import VApp from './VApp';
export { VApp };
export default VApp;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WQXBwL2luZGV4LmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBLE9BQU8sSUFBSSxNQUFNLFFBQVEsQ0FBQTtBQUV6QixPQUFPLEVBQUUsSUFBSSxFQUFFLENBQUE7QUFDZixlQUFlLElBQUksQ0FBQSIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBWQXBwIGZyb20gJy4vVkFwcCdcblxuZXhwb3J0IHsgVkFwcCB9XG5leHBvcnQgZGVmYXVsdCBWQXBwXG4iXX0=