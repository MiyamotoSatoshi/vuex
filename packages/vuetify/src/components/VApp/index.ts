import VApp from './VApp'

export { VApp }
