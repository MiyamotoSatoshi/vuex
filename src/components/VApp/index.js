import VApp from './VApp'

/* istanbul ignore next */
VApp.install = function install (Vue) {
  Vue.component(VApp.name, VApp)
}

export { VApp }
export default VApp
