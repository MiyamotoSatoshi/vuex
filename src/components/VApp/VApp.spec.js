import VApp from '~components/VApp'
import { test } from '~util/testing'

test('VApp.js', ({ mount }) => {
  it('should have an application class', () => {
    const wrapper = mount(VApp)

    expect(wrapper.hasClass('application')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have data-app attribute', () => {
    const wrapper = mount(VApp)

    expect(wrapper.getAttribute('data-app')).toBe('true')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should allow a custom id', () => {
    const wrapper = mount(VApp, {
      propsData: {
        id: 'inspire'
      }
    })

    expect(wrapper.getAttribute('id')).toBe('inspire')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
