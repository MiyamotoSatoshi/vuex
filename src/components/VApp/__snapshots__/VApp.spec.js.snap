// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VApp.js should allow a custom id 1`] = `

<div data-app="true"
     class="application application--light"
     id="inspire"
>
</div>

`;

exports[`VApp.js should have an application class 1`] = `

<div data-app="true"
     class="application application--light"
     id="app"
>
</div>

`;

exports[`VApp.js should have data-app attribute 1`] = `

<div data-app="true"
     class="application application--light"
     id="app"
>
</div>

`;
